src/main/java/com/faforever/client/chat/ChannelTabController.java

```diff
@@ -115,17 +123,33 @@ public void afterPropertiesSet() throws Exception {
   @Override
   public void initialize() {
     super.initialize();
-    JavaFxUtil.bindManagedToVisible(topicPane, chatUserList);
+    JavaFxUtil.bindManagedToVisible(topicPane, chatUserList, changeTopicTextButton, topicTextField,
+        acceptChangesTopicTextButton, cancelChangesTopicTextButton, topicText, topicCharactersLimitLabel);
+    JavaFxUtil.bind(topicText.visibleProperty(), topicTextField.visibleProperty().not());
+    JavaFxUtil.bind(topicCharactersLimitLabel.visibleProperty(), topicTextField.visibleProperty());
+    JavaFxUtil.bind(acceptChangesTopicTextButton.visibleProperty(), topicTextField.visibleProperty());
+    JavaFxUtil.bind(cancelChangesTopicTextButton.visibleProperty(), acceptChangesTopicTextButton.visibleProperty());
     JavaFxUtil.bind(chatMessageSearchTextField.visibleProperty(), chatMessageSearchContainer.visibleProperty());
     JavaFxUtil.bind(closeChatMessageSearchButton.visibleProperty(), chatMessageSearchContainer.visibleProperty());
     JavaFxUtil.bind(chatUserList.visibleProperty(), userListVisibilityToggleButton.selectedProperty());
 
     userListVisibilityToggleButton.setSelected(preferencesService.getPreferences().getChat().isPlayerListShown());
+    topicTextField.setTextFormatter(new TextFormatter<>(change -> change.getControlNewText().length() <= TOPIC_CHARACTERS_LIMIT ? change : null));
+
+    JavaFxUtil.addAndTriggerListener(topicTextField.textProperty(), new WeakInvalidationListener(changeTopicTextListener));
   }
 
   public void setChatChannel(ChatChannel chatChannel, BooleanBinding chatTabSelectedProperty) {
     this.chatChannel = chatChannel;
     this.channelName = chatChannel.getName();
+    Optional.ofNullable(chatChannel.getUser(userService.getUsername())).ifPresentOrElse(
+        user -> {
+          JavaFxUtil.bind(changeTopicTextButton.visibleProperty(), user.moderatorProperty().and(topicTextField.visibleProperty().not()));
```

Comment:
Can't use a bind here as moderatorProperty can be changed in any thread not just the application thread